### PR TITLE
Add pool protocol APIs

### PR DESCRIPTION
**Pull request by h9-dev**
Some codes from [pool-reference](https://github.com/Chia-Network/pool-reference.git).

- `get_pool_launcher`: Query initial state by `launcher_id`
- `get_pool_singleton_states`: Return join pool change history
- `get_pool_singleton_state`: Query last pool state by `launcher_id`
- `create_pool_absorb_tx`: Create absorb transaction.

### CODE REVIEW

File: chia/rpc/full_node_rpc_api.py [hunk 1]
```diff
@@ -577,3 +597,344 @@ async def get_mempool_item_by_tx_id(self, request: Dict) -> Optional[Dict]:
...
+                if not self._validate_pool_puzzle_hash(
+                    launcher_id,
+                    delay_puzzle_hash,
+                    delay_time,
+                    last_not_none_state,
+                    next_coin_record.coin.puzzle_hash,
+                    self.service.constants.GENESIS_CHALLENGE,
+                ):
+                    raise ValueError(f"Invalid singleton puzzle_hash for 0x{launcher_id}")
+                break
```
Comment (arvidn) [inline chia/rpc/full_node_rpc_api.py:826]: this whole block is copy pasted (from line 729). It should probably be turned into a function

**Review by arvidn — COMMENTED** [review]
without unit tests, can we know this works?

**Comment by mariano54** [comment]
Thank you for the PR!
While this is useful, I don't think it should be added to the official full node api. Pooling is a higher level thing which should be implemented outside of the full node.

**Comment by hpool-dev** [comment]
> Thank you for the PR!
> While this is useful, I don't think it should be added to the official full node api. Pooling is a higher level thing which should be implemented outside of the full node.

I think it's useful. Because a lot of dev use java, golang and nodejs to implement pool protocol.They can't use python's code directly, which would be more convenient if it could be made into RPCs.

**Pull request closed without merging**